datastore/providers/opensearch_datastore.py

```diff
@@ -0,0 +1,422 @@
+                                                                   conflicts="proceed")
+                print(f"Deletion Response from filters: {response}, {response.get('failures')}")
+                if not (response.get("failures") is not None and len(response['failures']) == 0):
+                    raise Exception(f"Error while deleting the documents using Filer: {filter}. Response: {response}")
+            except Exception as e:
+                print(f"Error deleting document for filters {delete_query_body} {e}")
+                raise e
+
+        # Delete by explicit ids
+        if ids:
+            try:
+                print(f"Deleting document for ids {ids}")
+                doc_ids_clause = []
+                # find all keys associated with the document ids
+                for document_id in ids:
+                    document_match_clause = {
+                        "match": {"metadata.document_id": document_id}
+                    }
+                    doc_ids_clause.append(document_match_clause)
+                delete_query_body = {
+                    "query": {
+                        "bool": {
+                            "should": doc_ids_clause
+                        }
+                    }
+                }
+                response = await self.async_client.delete_by_query(OPENSEARCH_INDEX_NAME, body=delete_query_body,
+                                                                   conflicts="proceed")
+                print(f"Deleted document Response from delete by document_id : {response}")
+            except Exception as e:
+                print(f"Error deleting ids: {ids} with exception : {e}")
+                raise e
+
+        return True
+
+    async def __setup_and_validate_data_store(self):
+        await self.__check_knn_plugin_present()
+        await self.__validate_index_setup()
+        await self.__validate_cluster_health()
+
+    async def __check_knn_plugin_present(self):
+        response = str(await self.async_client.cat.plugins())
+        if response.__contains__('opensearch-knn') is False:
+            raise Exception("The OpenSearch Data store doesn't contains the K-NN plugin. Please install K-NN plugin. "
+                            "Ref: https://opensearch.org/docs/latest/install-and-configure/plugins/ ")
+        print("OpenSearch K-NN plugin present in OpenSearch Cluster")
+
+    async def __validate_index_setup(self):
+        if (await self.async_client.indices.exists(OPENSEARCH_INDEX_NAME)) is True:
+            print(f"Index with name : {OPENSEARCH_INDEX_NAME}, already created.")
+            await self.__check_if_vector_field_present()
+        else:
+            await self.__create_index()
+
+    async def __check_if_vector_field_present(self):
+        mappings_response = str(await self.async_client.indices.get_mapping(index=OPENSEARCH_INDEX_NAME))
+        if mappings_response.__contains__("knn_vector") is False:
```

Comment:
Should we check field name here is mapped to knn_vector field?

Reply:
It's will be a complex check. Hence I added this very simple check.